sqlmesh/core/context.py

```diff
@@ -517,41 +532,68 @@ def state_reader(self) -> StateReader:
 
     def refresh(self) -> None:
         """Refresh all models that have been updated."""
-        if self._loader.reload_needed():
+        if (self._dbt_loader and self._dbt_loader.reload_needed()) or (
+            self._sqlmesh_loader and self._sqlmesh_loader.reload_needed()
+        ):
             self.load()
 
     def load(self, update_schemas: bool = True) -> GenericContext[C]:
         """Load all files in the context's path."""
         load_start_ts = time.perf_counter()
-        with sys_path(*self.configs):
-            project = self._loader.load(self, update_schemas)
-            self._macros = project.macros
-            self._jinja_macros = project.jinja_macros
-            self._models = project.models
-            self._metrics = project.metrics
-            self._standalone_audits.clear()
-            self._audits.clear()
+
+        projects = [
+            project
+            for loader, configs in [
+                (self._dbt_loader, self.dbt_configs),
+                (self._sqlmesh_loader, self.sqlmesh_configs),
+            ]
+            for project in [
+                self._load_factory("project", loader, configs, update_schemas=update_schemas)
+            ]
+            if isinstance(project, LoadedProject)
+        ]
+
+        self._standalone_audits.clear()
+        self._audits.clear()
+        self._macros.clear()
+        self._models.clear()
+        self._metrics.clear()
```

Comment:
Right it's because before when we had one project we were assigning to `_models`, `_metrics` etc, so if it already had values they were overwritten. Here since I chose to use update I had to clear the morels, macros and metrics before so that they were empty

Reply:
sorry i'm not sure i get it, why do you need to clear it?